elastic/kibana
# [Maps] adhoc data view support

Fixes https://github.com/elastic/kibana/issues/138140

To test from Lens:
* Install web logs sample data set
* Go to lens
* Open data view popover in top left. Click "Create a data view"
* Set index pattern to k* and select utc_time as timestamp.
* Click "Use without saving"
* In field list, click "geo.coordinates", then click "Visualize"
* Save map
* Reload map, verify adhoc data view works once map is reloaded.
* Verify adhoc data view state stored in maps saved object by viewing map saved object. Under mapStateJSON, there is new adHocDataViews array that contains data view spec.

To test from Discover:
* Install web logs sample data set
* Go to Discover
* Open data view popover in top left. Click "Create a data view"
* Set index pattern to k* and select utc_time as timestamp.
* Click "Use without saving"
* In field list, click "geo.coordinates", then click "Visualize"
* Save map
* Reload map, verify adhoc data view works once map is reloaded.
* Verify adhoc data view state stored in maps saved object by viewing map saved object. Under mapStateJSON, there is new adHocDataViews array that contains data view spec.
